controllers/vrg_volrep.go

```diff
@@ -1466,26 +1466,47 @@ func setPVCClusterDataProtectedCondition(protectedPVC *ramendrv1alpha1.Protected
 	}
 }
 
-// deleteVR deletes a VolumeReplication instance if found
-func (v *VRGInstance) deleteVR(vrNamespacedName types.NamespacedName, log logr.Logger) error {
+// ensureVRDeletedFromAPIServer adds an additional step to ensure that we wait for volumereplication deletion
+// from API server before moving ahead with vrg finalizer removal.
+func (v *VRGInstance) ensureVRDeletedFromAPIServer(vrNamespacedName types.NamespacedName, log logr.Logger) error {
 	cr := &volrep.VolumeReplication{
```

Comment:
Get just needs a pointer to store the fetched resource, and does not heed ObjectMeta defined, e.g see [this](https://github.com/RamenDR/ramen/blob/b0df036870caabee861ab18c1ea41e19da2d9248/controllers/vrg_volrep.go#L730-L736)